Add jumping rope, river and club

onaga1958 opened the pull request.

Commit:
Added jumping_rope and new test field

Commit:
updated jumping_rope

nzinov commented on game.py:
```diff
@@ -1,6 +1,6 @@
 from constants import DIRECTIONS
 from objects import OBJECTS
-from emoji import emojize
+#from emoji import emojize
```
Почему ты это закомментил?

onaga1958 replied:
У меня почему-то не было этой библиотеки, а смайлы мне для тестов были не нужны. Забыл раскоментить.

nzinov commented on game.py:
```diff
@@ -34,8 +34,10 @@ def move(self, game, direction):
                 game.field[game.player().position].arrive(game, game.player())
         if result:
             game.log(game.player(), emojize("Вы сходили {}".format(NAME[direction]), use_aliases=True))
+           #game.log(game.player(), "Вы сходили {}".format(NAME[direction]))
```
Это лучше убрать

nzinov commented on objects.py:
```diff
@@ -40,3 +40,33 @@ def action(game, player, action):
                     return True
+
+@register_object("скакалка")
+class Jumping_Rope(Object):
```
класс должен называться JumpingRope

nzinov commented on objects.py:
```diff
@@ -40,3 +40,33 @@ def action(game, player, action):
+    @staticmethod
+    def action(game, player, action):
+        if action == "прыгать":
+            game.log("Вы попрыгали. Кажется, под Вами и правда земля")
```
Лучше "вы" с маленькой буквы

nzinov commented on objects.py:
```diff
@@ -40,3 +40,33 @@ def action(game, player, action):
+                    if game.fields[other.field].can_move(other.position, direction):
+                        other.position = (other.position[0] + direction[0], other.position[1] + direction[1])
+                        game.fields[other.field].squares[other.position[0]][other.position[1]].arrive(game, other) 
+            if hit == False:
```
if not hit :smile:

onaga1958 replied:
А точно, лол)

onaga1958 committed:
Added Club

onaga1958 committed:
Added River and turn counter

nzinov commented on effects.py:
```diff
@@ -69,6 +71,9 @@ def event(self, game, player, event):
         if event == "start":
             self.spirit = deepcopy(player)
             self.spirit.effects.pop() #remove this effect
+            for eff in self.spirit.effects:
```
Это зачем?

onaga1958 replied:
Если реальное тело вступило в отложенный сон еще раз, не успев уснуть, то у спирита этот второй эффект тоже появиться. Если этот второй сон триггернется, то произойдет что-то странное.

nzinov replied:
А по-моему нормально - заснет во сне и все тут

onaga1958 replied:
Но тогда новый дух создаться в том же мире. Хз, мне кажется это все же странно. Тогда можно будет духом духа убить духа и дважды проснуться, лол. Хотя, это даже прикольно

nzinov replied:
Да, все равно ведь это все во сне.) Ну и это сохраняет ту логику, что во
сне с тобой происходит все то, что происходило бы наяву

nzinov commented on effects.py:
```diff
@@ -78,6 +83,24 @@ def event(self, game, player, event):
             self.expire(game, player)
             return False
 
+class DeferredSleep(ExpiringEffect):
```
Думаю, что лучше сделать отложенный эффект в общем случае, а не только для сна.

onaga1958 committed:
Try to add DeferredSleep. Now there is a bag: when u go to this square
in the second time, until the first effect expire, the second effect
will trigger one move later than it has to.

onaga1958 committed:
In the last commit forgot to delete these debugging strings

nzinov commented:
Да, все равно ведь это все во сне.) Ну и это сохраняет ту логику, что во
сне с тобой происходит все то, что происходило бы наяву

Николай Зинов
nzinov@gmail.com

вс, 9 окт. 2016 г. в 15:59, onaga1958 <notifications@github.com>:

> *@onaga1958* commented on this pull request.
> ------------------------------
>
> In effects.py <https://github.com/nzinov/maze/pull/1>:
>
> > @@ -69,6 +71,9 @@ def event(self, game, player, event):
>          if event == "start":
>              self.spirit = deepcopy(player)
>              self.spirit.effects.pop() #remove this effect
> +            for eff in self.spirit.effects:
>
> Но тогда новый дух создаться в том же мире. Хз, мне кажется это все же
> странно. Тогда можно будет духом духа убить духа и дважды проснуться, лол.
> Хотя, это даже прикольно
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/nzinov/maze/pull/1>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGF_lnFRSEDyRZ-K_FrzD5LwUCKVzoKvks5qyOU9gaJpZM4KRxav>
> .
>

onaga1958 committed:
Fixed remaning effects after death
